### PR TITLE
add interaction matrix method to register

### LINKED ISSUE #417: add interaction_matrix method to Register
add `interaction_matrix` method to Register

##  Description
From a given register, the `interaction_matrix` method should return a symmetric matrix encoding the Rydberg interactions $J_{ij}=1/r_{ij}^6$, with $r_{ij}=|r_{i}-r_{j}|$

### PR DESCRIPTION
**Pull request by sgrava**
Resolves #417 

> add `interaction_matrix` method to Register
> 
> ##  Description
> From a given register, the `interaction_matrix` method should return a symmetric matrix encoding the Rydberg interactions $J_{ij}=1/r_{ij}^6$, with $r_{ij}=|r_{i}-r_{j}|$

## Changes
- add `Register.interaction_matrix` method which return the Rydberg interaction matrix. Dispatches to `torch` or `numpy` depending on the coordinates type stored in the register
- test

## Bonus
- adds `matplotlib.use("Agg")` in `conftest.py`. Totally unrelated but let the tests to avoid loading matplotlib in interacting mode. Saves 0.5 seconds per test module.

## Next
- refactor abstract array interface #406

### CODE REVIEW

**Review by v-vitale — COMMENTED** [review]
Just a comment.

What is the reason behind using torch tensors for the interaction matrix?

**Comment by sgrava** [comment]
> Just a comment.
> 
> What is the reason behind using torch tensors for the interaction matrix?

It would be nice to support torch backed registers as well, but not a priority for now I agree

File: qoolqit/register.py [hunk 1]
```diff
@@ -310,6 +317,22 @@ def interactions(self) -> dict:
...
+    def interaction_matrix(self) -> npt.NDArray[np.float64] | torch.Tensor:
+        """Interaction 1/r^6 between each qubit pair, as a matrix (0 on the diagonal)."""
+        dist_matrix = _pdist(self._coords)
+
+        # Avoid division-by-zero on the diagonal (where r_ii == 0):
+        # - torch: use a boolean diagonal mask and `torch.where` (out-of-place, autograd-safe).
+        # - numpy: set dist_matrix diagonal to 1.0 before exponentiation, then back to 0.0.
+        if _is_torch(dist_matrix):
+            diagonal_mask = torch.eye(dist_matrix.shape[0], dtype=torch.bool)
+            return torch.where(diagonal_mask, 0.0, dist_matrix ** (-6))
```
Comment (abussy-pasqal) [inline qoolqit/register.py]: no need to mask, dividing by zero with torch won't emit a warning

Comment (sgrava) [reply qoolqit/register.py]: Yes but it is more really to sanitize the output and return 0 on the diagonal by convention.
The `torch.fill_diagonal_` method is not really an alternative since it writes in-place.

Do you see an alternative here? Keeping the convention to have  0 on the diagonal I mean

Comment (abussy-pasqal) [reply qoolqit/register.py]: you can fill the diagonal with zeros after the computation

**Review by abussy-pasqal — DISMISSED** [review]

File: qoolqit/register.py (hunk 1 above)
Comment (sgrava) [reply qoolqit/register.py]: I think it is more clear now that filling the diagonal of a torch tensor needs to be out-of-place. Please, have another look!

**Review by abussy-pasqal — APPROVED** [review]
